### ISSUE TITLE
no-mutating-props seems too strict

### ISSUE THREAD

**Issue by leosdad** [vuejs/eslint-plugin-vue#1314]
**What rule do you want to change?**

no-mutating-props

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

A new option would suffice, but maybe the default behavior should also be changed as well.

**Please provide some example code that this change will affect:**

```vue
<script>
props: {
  localData: { type: Object, required: true },
},
created()
{
  if(!this.localData.screens) {
    this.localData.screens = {};
  }
}
</script>
```
**What does the rule currently do for this code?**

It flags it as an error, and I have to disable it globally or locally.

**What will the rule do after it's changed?**

This should be allowed because the property is an object. I'm not mutating `this.localData` itself, but another object inside it.

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for posting this issue.

Changing the state of a parent component passed by reference on a child component can be confusing.
I recommend that you use an event to communicate from a child to a parent.
https://vuejs.org/v2/style-guide/index.html#Implicit-parent-child-communication-use-with-caution

I don't want to change this rule because I think it should be strict.
You are free to turn this rule off in your configuration.

Thank you for your understanding.

**Comment by pimlie** [comment]
@ota-meshi Sorry for the duplicate issue, seems like Github search returns different results for _vue/no-mutating-props_ when searching the whole repo and when searching just issues (which I did)

Could we maybe toggle this strict behaviour with a rule option? _Priority D / Use with caution_ doesnt mean its plain wrong and I would love to be able to still make sure that I didn't really overwrite a prop somewhere.

**Comment by decademoon** [comment]
Is there any way to disallow `v-model="value"` (where `value` is a prop) but still allow `v-model="value.foo"`?
